### PR TITLE
Added new GPIO commands

### PR DESCRIPTION
**Pull request by shaggygi**
Added a couple new GPIO commands.

Renamed and random cleaning of other related commands.

Cleaned up the DeviceApiTester README while updating command references, as well.

### CODE REVIEW

**Comment by joperezr** [comment]
> // Licensed to the .NET Foundation under one or more agreements.

Did anything else other than the fileRename changed here?

---
Refers to: tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs:1 in 056894a. [](commit_id = 056894a9ac17b9d1b0cc071803f88531cecb7369, deletion_comment = False)

File: tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                    // Open the GPIO pin connected to the LED if one was specified.
+                    if (LedPin != null)
+                    {
+                        gpio.OpenPin(LedPin.Value, PinMode.Output);
+                        gpio.Write(LedPin.Value, OffValue);
+                    }
+
+                    // Declare a local function to handle the pin value changed events.
+                    int count = 0;
+                    void valueChangeHandler(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
```
Comment (joperezr) [inline tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs]: I know you didn't add this local function, but could you please push it to the end of this method j ust like the other local function? Sometimes it's hard to read these, so usually having them after the return statement of the function makes it a bit more readable.

File: tools/DevicesApiTester/Commands/Gpio/GpioReadPin.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        /// <remarks>
+        ///     NOTE: This test app uses the base class's <see cref="CreateGpioController"/> method to create a device.<br/>
+        ///     Real-world usage would simply create an instance of <see cref="GpioController"/>:
+        ///     <code>using (var gpio = new GpioController())</code>
+        /// </remarks>
+        public int Execute()
+        {
+            Console.WriteLine($"Driver={Driver}, Scheme={Scheme}, Pin={Pin}");
+
+            using (GpioController gpio = CreateGpioController())
```
Comment (joperezr) [inline tools/DevicesApiTester/Commands/Gpio/GpioReadPin.cs]: NIT: I liked how you renamed this to be `controller` instead, can we do this here as well?

File: tools/DevicesApiTester/Commands/Gpio/GpioWritePin.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        /// <remarks>
+        ///     NOTE: This test app uses the base class's <see cref="CreateGpioController"/> method to create a device.<br/>
+        ///     Real-world usage would simply create an instance of <see cref="GpioController"/>:
+        ///     <code>using (var gpio = new GpioController())</code>
+        /// </remarks>
+        public int Execute()
+        {
+            Console.WriteLine($"Driver={Driver}, Scheme={Scheme}, Pin={Pin}, Value={Value}");
+
+            using (GpioController gpio = CreateGpioController())
```
Comment (joperezr) [inline tools/DevicesApiTester/Commands/Gpio/GpioWritePin.cs]: Same NIT here for the name.

File: tools/DevicesApiTester/Commands/Gpio/GpioBlinkLed.cs [hunk 4]
```diff
@@ -1,67 +1,67 @@
...
+using System;
+using System.Device.Gpio;
+using System.Threading.Tasks;
+using CommandLine;
+using DeviceApiTester.Infrastructure;
+
+namespace DeviceApiTester.Commands.Gpio
+{
+    [Verb("gpio-blink-led", HelpText = "Blinks an LED connected to a specified GPIO pin.")]
+    public class GpioBlinkLed : GpioCommand, ICommandVerbAsync
```
Comment (joperezr) [inline tools/DevicesApiTester/Commands/Gpio/GpioBlinkLed.cs]: These commands are already within a namespace that contains Gpio, so I'm not sure if the extra Gpio on the name makes sense or is a bit overkill IMO

File: tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                    cancelEvent.WaitOne();
+
+                    gpio.ClosePin(ButtonPin);
+                    if (LedPin != null)
+                    {
+                        gpio.ClosePin(LedPin.Value);
+                    }
+
+                    // Unregister the event handler for changes to the pin value
+                    gpio.UnregisterCallbackForPinValueChangedEvent(ButtonPin, valueChangeHandler);
```
Comment (buyaa-n) [inline tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs]: This test could be more demonstrative by covering/using more than one event. For example this is my event handler test I used, it is not perfect but covering more test scenarios, it uses 2 buttons though, just to give you some other idea:
```csharp
ManualResetEvent mre = new ManualResetEvent(false);
                using (GpioController controller = new GpioController(PinNumberingScheme.Logical , UnixDriver.InitUnixDriver()))
                {
                    Console.WriteLine("event registration test: " );
                    PinValue ledValue = PinValue.Low;
                    controller.OpenPin(23, PinMode.Output);
                    controller.OpenPin(18, PinMode.Input);
                    controller.OpenPin(25, PinMode.Input);
                    int clickCount = 0;
                    controller.RegisterCallbackForPinValueChangedEvent(18, PinEventTypes.Rising, (o, e) => {
                        Console.WriteLine("Raising event 1 called");
                        ledValue = toggleValue(ledValue);
                        controller.Write(23, ledValue);
                        clickCount ++;
                            if (clickCount > 10) {
                                mre.Set();
                            }
                        });  
                    PinChangeEventHandler callback = (o, e) => { 
                        Console.WriteLine("Raising event 2 delegate for remove called");
                    };
                    controller.RegisterCallbackForPinValueChangedEvent(18, PinEventTypes.Rising, callback);
                    controller.RegisterCallbackForPinValueChangedEvent(18, PinEventTypes.Rising, (o, e) => {
                        Console.WriteLine("Raising event 3 called");
                    });

                    controller.RegisterCallbackForPinValueChangedEvent(18, PinEventTypes.Falling, (o, e) => {
                        Console.WriteLine("Falling event 1 called");
                    });
                    controller.RegisterCallbackForPinValueChangedEvent(25, PinEventTypes.Rising, (o, e) => {
                        Console.WriteLine("Remove callback event occured");
                        controller.UnregisterCallbackForPinValueChangedEvent(18, callback);
                    });
                    mre.WaitOne();
                }
```

File: tools/DevicesApiTester/Commands/Gpio/GpioBlinkLed.cs (hunk 4 above)
Comment (shaggygi) [reply tools/DevicesApiTester/Commands/Gpio/GpioBlinkLed.cs]: It's a little overkill, but trying to make it consistent with other commands (I2c, Spi, etc.) and it matches the actual command string ("gpio-blink-led" in this case).

File: tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs (hunk 5 above)
Comment (shaggygi) [reply tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs]: Thanks for the feedback @buyaa-n.  I have not studied the events yet as that is next up on part of my Mcp23xxx binding exercise.  Let's keep as is for now and maybe circle back if needed.

File: tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs (hunk 1 above)
Comment (shaggygi) [reply tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs]: This was mainly a file rename.  I reordered the display of properties for better flow, but nothing logic -wise changed.

File: tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs (hunk 5 above)
Comment (buyaa-n) [reply tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs]: Sounds good, thanks

**Pull request merged**